src/caStatusBarProvider.ts

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+import { StatusBarItem, window, StatusBarAlignment } from "vscode";
+import { Disposable } from "vscode-languageclient";
+import { StatusMessages } from "./statusMessages";
+import { Commands } from "./commands";
+
+class CAStatusBarProvider implements Disposable {
+    private statusBarItem: StatusBarItem;
+
+    constructor() {
+        this.statusBarItem = window.createStatusBarItem(StatusBarAlignment.Left, 0);
+    }
+
+    public showSummary(text: string): void {
+        this.statusBarItem.text = text;
+        this.statusBarItem.command = {
+            title: StatusMessages.FULL_STACK_PROMPT_BUTTON,
+            command: Commands.TRIGGER_FULL_STACK_ANALYSIS,
+        };
+        this.statusBarItem.tooltip = StatusMessages.FULL_STACK_PROMPT_BUTTON;
+        this.statusBarItem.show();
+    }
+
+    public setError(): void {
+        this.statusBarItem.text = `$(thumbsdown) analysis failed due to error`;
+        this.statusBarItem.command = {
+            title: StatusMessages.FULL_STACK_PROMPT_BUTTON,
+            command: Commands.TRIGGER_STACK_LOGS,
+        };
+    }
+
+    public dispose(): void {
+        this.statusBarItem.dispose();
+    }
+}
+
+export const caStatusBarProvider: CAStatusBarProvider = new CAStatusBarProvider();
```

Comment:
we should simply import it and create an instance wherever required WTYT?

Reply:
It is a singleton instance, there are no use case to have multiple instance of this.

Reply:
Okay, then it should be fine..